ansible/roles/cli/tasks/deploy.yml

```diff
@@ -3,26 +3,19 @@
 
 - name: "ensure nginx directory for cli exists"
   file:
-    path: "{{ cli_nginx_dir  }}"
+    path: "{{ cli_nginx_dir }}"
     state: directory
 
-- include: docker_login.yml
+- set_fact:
+    cli_installation_mode="{{ openwhisk_cli.installation_mode }}"
 
-- name: "pull the {{ docker_image_tag }} image of cli"
-  shell: "docker pull {{ docker_registry }}{{ docker_image_prefix }}/cli:{{ docker_image_tag }}"
-  when: docker_registry != ""
+- name: "download the binaries from the release page of OpenWhisk CLI"
+  include: download_openwhisk_cli.yml
+  with_items: "{{ cli_os }}"
+  when: cli_installation_mode == "download"
 
-- name: "run Docker"
-  shell: "docker run --name cli {{ docker_registry }}{{ docker_image_prefix }}/cli:{{ docker_image_tag }}"
-
-- name: "copy all the cli binaries into nginx"
-  shell: "docker cp cli:/src/github.com/go-whisk-cli/build/. '{{ cli_nginx_dir }}'"
-  ignore_errors: yes
-
-- name: "stop Docker"
-  shell: "docker stop cli"
-
-- name: "remove cli container after publishing"
-  shell: "docker rm cli"
+- name: "download the source code if necessary and build the binaries locally"
```

Comment:
here source code is still downloaded from https://github.com/openwhisk/openwhisk-cli.git instead of considering local copy of openwhisk/openwhisk-cli, isn't it? What's difference between downloading source from repo and than building it locally v/s downloading binaries which are built and available on github, in terms of source code within the binaries?